nilearn/image/image.py

```diff
@@ -205,18 +208,25 @@ def _smooth_array(arr, affine, fwhm=None, ensure_finite=True, copy=True):
         # SPM tends to put NaNs in the data outside the brain
         arr[np.logical_not(np.isfinite(arr))] = 0
 
-    if fwhm == 'fast':
+    if isinstance(fwhm, str) and (fwhm == 'fast'):
         arr = _fast_smooth_array(arr)
     elif fwhm is not None:
+        # Assume fwhm is either a nonzero number or a tuple, or a numpy array.
+        # Whatever it is at this point, transform it into an np.array
+        fwhm = np.asarray(fwhm)
+        fwhm = np.where(fwhm == None, 0.0, fwhm)
+
         # Keep only the scale part.
         affine = affine[:3, :3]
 
         # Convert from a FWHM to a sigma:
         fwhm_over_sigma_ratio = np.sqrt(8 * np.log(2))
         vox_size = np.sqrt(np.sum(affine ** 2, axis=0))
         sigma = fwhm / (fwhm_over_sigma_ratio * vox_size)
+
         for n, s in enumerate(sigma):
-            ndimage.gaussian_filter1d(arr, s, output=arr, axis=n)
+            if s > 0.0:
```

Comment:
Generally this is not a good way to compare floats. If `s == 0.00045` this will be `True` which may or may not be the intention. Generally floats are compared by `abs(s-0.0) > epsilon`, where the epsilon value is set according to desired sensitivity. One can also use `math.isclose()` or `numpy.isclose()`.

@jeromedockes WDYT?

Reply:
this is true, and we could also skip the whole conversion of None to 0 and then skipping smoothing in dimensions where sigma == 0 however smoothing with a very small sigma and not smoothing is the same; the only important thing is to make sure we don't pass a negative number or exactly 0 to gaussian_filter1d as that would cause an error